src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1274,6 +1279,12 @@ internal async ValueTask<NpgsqlDataReader> ExecuteReader(CommandBehavior behavio
                             await reader.NextResultAsync(finalCt);
                         else
                             reader.NextResult();
+
+                        if (Log.IsEnabled(NpgsqlLogLevel.Debug))
```

Comment:
This is still an added check to the hot path, which I'm not very happy about - but it's going to be hard to get rid of that. We could put all the code between the start and end in a local function, and call that either with or without the stopwatch. But that's a bit extreme, I will try to measure the perf impact of this on the TechEmpower platform.

Reply:
@roji I thought about putting all of this in a local function and call it with or w/o the stopwatch but I thought that it would be an overkill. I am not really an expert on this, but I am curious to see the perf impact this has.

Reply:
I'm not convinced it's not overkill either... But in any case we have the [outstanding design question](https://github.com/npgsql/npgsql/issues/2282#issuecomment-699949166) which may make this question moot.